File: lib/py/inline.py
## @main
if __name__ == "__main__":
	main()

## @error;@SoftwareFault
def libbun_error(msg):
	raise SoftwareFault(msg)

## @Fault
class Fault:
	def __init__(self, msg):
		self.msg = msg
	def __str__(self):
		return self.msg

## @SoftwareFault;@Fault
class SoftwareFault(Fault):
	def __init__(self, msg):
		self.msg = unicode(msg)

## @null
def libbun_null(s) : 
	return s if s != None else u'null'

## @catch;@SoftwareFault
def libbun_catch(e):
	if isinstance(e, Fault):
		return e
	if isinstance(e, ZeroDivisionError):
		return SoftwareFault(str(e))
	if isinstance(e, KeyError):
		return SoftwareFault('key not found: "%s"' % e)
	if isinstance(e, IndexError):
		return SoftwareFault(str(e))
	return Fault(str(e))

## @arraysize
def libbun_arraysize(a, n, v):
	while len(a) < n: a.append(v)
	while len(a) > n: a.pop()

## @mapget
def libbun_mapget(m, k, v):
	if m.has_key(k): return m[k]
	else: return v

File: lib/js/inline.js
// @main
if(main){
	main();
}

// @extend
var __extends = this.__extends || function (d, b) {
	for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
	function __() { this.constructor = d; }
	__.prototype = b.prototype;
	d.prototype = new __();
}

// @Fault
var Fault = (function () {
    function Fault(message) {
        this.message = message;
        this.name = "Fault";
    }
    return Fault;
})();

// @SoftwareFault;@Fault;@extend
var SoftwareFault = (function (_super) {
    __extends(SoftwareFault, _super);
    function SoftwareFault(message) {
        _super.call(this, message);
        this.name = "SoftwareFault";
    }
    return SoftwareFault;
})(Fault);

// @catch;@SoftwareFault
function libbun_catch(e){
	return e;
}

// @strnull
function libbun_strnull(s){
	return s ? s : "null";
}

// @arraysize
def libbun_arraysize(a, n, v):
	while len(a) < n: a.append(v)
	while len(a) > n: a.pop()

// @mapget;@SoftwareFault
function libbun_mapget(m, k, v){
    if(k in m) return m[k];
    if(v) return v;
    throw new SoftwareFault("key not found: " + k);
}

// @arrayget;@SoftwareFault
function libbun_arrayget(a, i){
	if(i >= 0 && i < a.length) return a[i];
	throw new SoftwareFault("out of index: " + i);
}

// @intdiv
function libbun_intdiv(a, b){
	if(b == 0) throw new SoftwareFault("zero division");
	return Math.floor(a / b);
}
